src/gui/TabManager.cc

```diff
@@ -485,23 +483,7 @@ void TabManager::openTabFile(const QString& filename)
     setTabName(nullptr);
     editor->setPlainText(cmd.arg(filename));
   }
-  par->fileChangedOnDisk();   // force cached autoReloadId to update
-  bool opened = refreshDocument();
-
-  if (opened) {   // only try to parse if the file opened
-    par->hideCurrentOutput();     // Initial parse for customizer, hide any errors to avoid duplication
-    try {
-      par->parseTopLevelDocument();
-    } catch (const HardWarningException&) {
-      par->exceptionCleanup();
-    } catch (const std::exception& ex) {
-      par->UnknownExceptionCleanup(ex.what());
-    } catch (...) {
-      par->UnknownExceptionCleanup();
-    }
-    par->lastCompiledDoc = ""; // undo the damage so F4 works
-    par->clearCurrentOutput();
-  }
```

Comment:
This part is manipulating the MainWindow  to keep its internal state synchronized instead of relying on signals from TabManager to MainWindow slots. Not sure I re-implemented the complete behavior in MainWindow, if some of you have idea on possible side effect, please tell me.

Reply:
The purpose of this code is to initialize the Customizer window without first having to press preview or render. That feature wasn't ported back to MainWindow.

Reply:
Hi Marius

I was thinking the initizialization of the Customizer was properly done in the slot: MainWindow::onTabManagerEditorChanged(EditorInterface *newEditor)
The slot is connected to the TabManager signal's currentEditorChanged() that is emitted when a tab is opened. 

I will give it a look to see what is missing.

Reply:
So I have added a MainWindow::onTabMangerEditorCreated signal so at creation of a file, even if it is not previewed its content is parsed so the Customizer has the proper content.